app/models/post.rb

```diff
@@ -0,0 +1,23 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', counter_cache: true
+  validates_presence_of :author
+  has_many :likes, foreign_key: :post_id
+  has_many :comments, foreign_key: :post_id
+
+  after_create :increment_author_posts_counter
+  after_destroy :decrement_author_posts_counter
+
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  private
+
+  def increment_author_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def decrement_author_posts_counter
+    author.decrement!(:posts_counter)
+  end
```

Comment:
- [ OPTIONAL] You did an amazing job, to create the increment and decrement methods,  but What I what I would recommend is to use `after_save `than `after_create `to create the callback functions in all your tables's methods,  because `after_create `is triggered only when a new object is created and saved for the first time. it will not be used for an existing record.

Reply:
Implemented suggestion change after_create to after_save in post.rb✔